tock/hours/models.py

```diff
@@ -218,14 +218,22 @@ def save(self, *args, **kwargs):
         super().save(*args, **kwargs)
 
     @classmethod
-    def get_most_recent_periods(cls, number_of_periods=1):
+    def get_most_recent_periods(cls, number_of_periods=1, number_of_results=None):
         """
         Return the most recent N completed reporting periods
         A reporting period is complete if it's end date
         has is today or in the past.
         """
         today = datetime.date.today()
-        return ReportingPeriod.objects.filter(end_date__lte=today).order_by('-start_date')[:number_of_periods]
+        periods = ReportingPeriod.objects.filter(end_date__lte=today)
+
+        if number_of_periods and number_of_results:
+            starting_index = number_of_periods - 1
+            range_end = starting_index + number_of_results
+            return periods.order_by('-start_date')[starting_index:range_end]
```

Comment:
Noted above but this will break for any input where the `starting_index` is larger than the number of reporting periods in the system.

Reply:
I believe this is resolved as well